### PR TITLE
Fix array_merge increment problems

### PR DESCRIPTION
**Pull request by ArrayIterator**
Document transformer transformMetaMembers() will reset array keys to increment value / integer when meta contains numeric float|int|number(int)string offset.

https://github.com/woohoolabs/yin/issues/108

### CODE REVIEW

File: src/JsonApi/Transformer/DocumentTransformer.php [hunk 1]
```diff
@@ -74,8 +74,12 @@ private function transformMetaMembers(AbstractDocumentTransformation $transforma
             $transformation->result["jsonapi"] = $jsonApi->transform();
         }
 
-        $meta = array_merge($transformation->document->getMeta(), $transformation->additionalMeta);
```
Comment (kocsismate) [inline src/JsonApi/Transformer/DocumentTransformer.php:77]: Please remove the unused `array_merge` usage at the beginning of the file, as it fails the CS job: https://github.com/woohoolabs/yin/actions/runs/3558742766/jobs/5979406267#step:7:18

**Review by kocsismate — COMMENTED** [review]
Looks good except of the coding style nit! If you have time, an additional unit test would also be welcome, but I can accept the PR without it as well.

File: src/JsonApi/Transformer/DocumentTransformer.php (hunk 1 above)
Comment (ArrayIterator) [reply src/JsonApi/Transformer/DocumentTransformer.php:77]: Ah I see .. just push first
Sorry for my mistake

Comment (kocsismate) [reply src/JsonApi/Transformer/DocumentTransformer.php:77]: No problem! :)

**Comment by kocsismate** [comment]
Thank you for your PR!

**Pull request merged**